File: chapter 1/ex_1_7.php
<?php

function average($x, $y)
{
    return ($x + $y) / 2;
}

function improve($guess, $x)
{
    return average($guess, ($x / $guess));
}

function smallEnough($x)
{
    return $x / 100;
}

function goodEnough($guess, $x)
{
    return abs(($guess ** 2) - $x) < smallEnough($x);
}

function sqrtIter($guess, $x)
{
    return goodEnough($guess, $x) ? $guess : sqrtIter(improve($guess, $x), $x);
}

echo sqrtIter(1, 9) . PHP_EOL;

File: chapter 1/ex_1_7.rb
def average(x, y)
  return (x + y) / 2.0
end

def improve(guess, x)
  return average(guess, (x / guess))
end

def smallEnough(x)
  return x / 100.0
end

def goodEnough(guess, x)
  return ((guess ** 2) - x).abs < smallEnough(x)
end

def sqrtIter(guess, x)
  return goodEnough(guess, x) ? guess : sqrtIter(improve(guess, x), x)
end

puts sqrtIter(1, 9)
